src/gui/core/event/distributor.cpp

```diff
@@ -316,6 +344,14 @@ void mouse_motion::mouse_leave()
 void mouse_motion::start_hover_timer(widget* widget, const point& coordinate)
 {
 	assert(widget);
+
+#ifdef __IPHONEOS__
+	// Guessing a crash location in a nasty stack in gui2::execute_timer.
+	// Either this or a long-touch menu.
+	// Remove this when the crash in gui2::execute_timer() and gui2::timer_callback() is gone and try again.
```

Comment:
aren't the mutexes you added in Timer.cpp supposed to fix this, or am I missing something?